File: Andorid/NewWeather/NewWeather/app/src/main/java/com/example/newweather/ListViewAdapter.java
package com.example.newweather;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

public class ListViewAdapter extends BaseAdapter {
    static class ViewHolder{
        TextView tvDate;
        TextView tvHigh;
        TextView tvLow;
        TextView weatherType;
        //mageView iv;
    }
    Context context;
    ArrayList<WeatherBean> arrayList;

    public ListViewAdapter(Context context, ArrayList<WeatherBean> arrayList) {
        this.context = context;
        this.arrayList = arrayList;
    }

    @Override
    public int getCount() {
        return arrayList.size();
    }

    @Override
    public Object getItem(int i) {
        return arrayList.get(i);
    }

    @Override
    public long getItemId(int i) {
        return i;
    }

    @Override
    public View getView(int i, View view, ViewGroup viewGroup) {
        ViewHolder viewHolder;
        LayoutInflater layoutInflater=LayoutInflater.from(context);
        if (view==null){
            view=layoutInflater.inflate(R.layout.line_item,viewGroup,false);
            viewHolder= new ViewHolder();
            viewHolder.tvDate=view.findViewById(R.id.tvDate);
            viewHolder.tvHigh=view.findViewById(R.id.tvHigh);
            viewHolder.tvLow=view.findViewById(R.id.tvLow);
            viewHolder.weatherType=view.findViewById(R.id.weatherType);
            //viewHolder.iv=view.findViewById(R.id.weatherType);
            view.setTag(viewHolder);
        } else {
            viewHolder= (ViewHolder) view.getTag();
        }
        String date = arrayList.get(i).getDate();
        date = date.substring(4,6)+"-"+date.substring(6,8);
        viewHolder.tvDate.setText(date);
        viewHolder.tvHigh.setText(arrayList.get(i).getHigh()+" °C");
        viewHolder.tvLow.setText(arrayList.get(i).getLow()+" °C");
        viewHolder.weatherType.setText(arrayList.get(i).getStatus());

        /*
        添加天气图片
        WeatherBean weatherBean=arrayList.get(i);
        String t=weatherBean.getStatus();
        switch (t) {
            case "雨":
                viewHolder.iv.setImageResource(R.drawable.rain);
                break;
            case "晴":
                viewHolder.iv.setImageResource(R.drawable.sun);
                break;
            case "雪":
                viewHolder.iv.setImageResource(R.drawable.snow);
                break;
            default:
                viewHolder.iv.setImageResource(R.drawable.snow);
                //throw new IllegalStateException("Unexpected value: " + t);
        }
         */
        return view;
    }
}

File: NewWeather/app/src/main/java/com/example/newweather/ListViewAdapter.kt
package com.example.newweather

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView


class ListViewAdapter(var context: Context, var arrayList: ArrayList<WeatherBean>) : BaseAdapter(){
    class ViewHolder{
        var tvDate: TextView? = null
        var tvLow: TextView? = null
        var tvHigh: TextView? = null
        var weatherType: TextView? = null
    }
    
    override fun getCount(): Int {
        return arrayList.size
    }

    override fun getItem(p0: Int): Any {
        return arrayList[p0]
    }

    override fun getItemId(p0: Int): Long {
        return p0.toLong()
    }

    override fun getView(i: Int, p1: View?, p2: ViewGroup?): View {
        var view: View?
        var viewHolder: ViewHolder
        if (p1 == null){
            view = LayoutInflater.from(context).inflate(R.layout.line_item, p2, false)
            viewHolder = ViewHolder()
            viewHolder.tvDate = view.findViewById(R.id.tvDate)
            viewHolder.tvLow = view.findViewById(R.id.tvLow)
            viewHolder.tvHigh = view.findViewById(R.id.tvHigh)
            viewHolder.weatherType = view.findViewById(R.id.weatherType)
            view.tag = viewHolder
        }else{
            view = p1
            viewHolder = view.tag as ViewHolder
        }
        viewHolder.tvDate?.text = arrayList[i].date
        viewHolder.tvLow?.text = arrayList[i].low
        viewHolder.tvHigh?.text = arrayList[i].high
        viewHolder.weatherType?.text = arrayList[i].status
        var date = arrayList[i].date
        var month = date.substring(4, 6)
        var day = date.substring(6, 8)
        if (month[0] == '0') month = String.format(" %s", month.substring(1))
        if (day[0] == '0') day = String.format(" %s", day.substring(1))
        date = month + '月' + day + '日'

        viewHolder.tvDate!!.text = date
        viewHolder.tvHigh!!.text = String.format("%s °C", arrayList[i].high)
        viewHolder.tvLow!!.text = String.format("%s °C", arrayList[i].low)
        viewHolder.weatherType!!.text = arrayList[i].status
        return view!!
    }
}